src/main/resources/META-INF/rewrite/apache-httpclient-5.yml

```
@@ -626,3 +639,31 @@ recipeList:
   - org.openrewrite.java.ChangeType:
       oldFullyQualifiedTypeName: org.apache.http.client.CredentialsProvider
       newFullyQualifiedTypeName: org.apache.hc.client5.http.auth.CredentialsStore
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.apache.httpclient5.MigrateBasicSchemeFactory
+displayName: Removes the charset argument from `BasicSchemeFactory()` constructor calls
+description: RFC 7617 mandates the use of UTF-8. This constructor is now deprecated and httpclient v5 ignores any provided charset, ensuring that only UTF-8 is used.
+preconditions:
+  - org.openrewrite.java.search.FindMethods:
+      methodPattern: org.apache.hc.client5.http.impl.auth.BasicSchemeFactory <constructor>(java.nio.charset.Charset)
+
+recipeList:
+  - org.openrewrite.java.DeleteMethodArgument:
+      methodPattern: org.apache.hc.client5.http.impl.auth.BasicSchemeFactory <constructor>(java.nio.charset.Charset)
+      argumentIndex: 0
+  - org.openrewrite.java.RemoveUnusedImports
```

I think we should adjust `DeleteMethodArgument` to call `maybeRemoveImport` on the type of the removed argument, such that we do not need to call `RemoveUnusedImports` here, or above.
```suggestion
  - org.openrewrite.java.DeleteMethodArgument:
      methodPattern: org.apache.hc.client5.http.impl.auth.BasicSchemeFactory <constructor>(java.nio.charset.Charset)
      argumentIndex: 0
```

---

That then also removes the need for preconditions here, so we can remove the wrapping declarative recipes.

---

Here's a quick test of what we're looking for from `DeleteMethodArgument`.
```java
@Test
void removeUnusedImportOfGenericType() {
    rewriteRun(
      spec -> spec.recipes(new DeleteMethodArgument("C foo(java.util.List)", 0)),
      java(
        """
          import java.util.List;
          class C {
             public static void foo(List<?> list) {}
          }
          """
      ),
      java(
        """
          import java.util.ArrayList;
          class A {{ C.foo(new ArrayList<>()); }}
          """,
        """
          class A {{ C.foo(); }}
          """
      )
    );
}
```

We'll also ofcourse want to similarly handle arguments that are a `Charset`, or field access on `Charsets`.